src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -112,14 +112,9 @@ public SemanticModel(Compilation compilation, BicepSourceFile sourceFile, IFileR
                 return parameters.ToImmutable();
             });
 
-            this.exportedTypesLazy = new(() => Root.TypeDeclarations.DistinctBy(t => t.Name)
-                // skip over any type without an `@export()` decorator
-                .Where(t => SemanticModelHelper.TryGetDecoratorInNamespace(this,
-                    t.DeclaringType,
-                    SystemNamespaceType.BuiltInName,
-                    LanguageConstants.ExportPropertyName) is not null)
-                .ToImmutableDictionary(t => t.Name,
-                    t => new ExportedTypeMetadata(t.Name, t.Type, DescriptionHelper.TryGetFromDecorator(this, t.DeclaringType))));
+            this.exportsLazy = new(() => ExportedTypes.Concat(ExportedVariables)
+                .DistinctBy(export => export.Name)
```

Comment:
FYI we've been trying to stick to the convention of using `LanguageConstants.IdentifierComparer` to be explicit about identifier comparisons. In fairness I don't know if that's really valuable at this point - we're never going to change the comparer we use without breaking the world.

Might be worth a discussion on whether we should just refactor to remove it - having a mix of conventions is probably the worst option!